File: LogicalThinkingPatterns/InvertedNumberedTrianglePattern/InvertedNumberTriangle.py
class InvertedNumberedTriangle:
    def printPattern(self, n):
        for i in range(n, 0, -1):
            pattern = ""
            for j in range(1, i+1):
                pattern += str(j)+" "
            print(pattern)

    def run(self):
        n = int(input("Enter the number of rows: "))
        self.printPattern(n)

triangle = InvertedNumberedTriangle()
triangle.run()

File: LogicalThinkingPatterns/InvertedNumberedTrianglePattern/InvertedNumberTriangle.ts
class InvertedNumberedTriangle {
  public printPattern(n: number): void {
    for (let i = n; i > 0; i--) {
      let pattern: string = "";
      for (let j = 1; j <= i; j++) {
        pattern += j + " ";
      }
      console.log(pattern);
    }
  }
}

const triangle = new InvertedNumberedTriangle();
triangle.printPattern(5);
